controllers/ibu_controller.go

```diff
@@ -99,40 +103,38 @@ func requeueWithCustomInterval(interval time.Duration) ctrl.Result {
 //
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
-func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (nextReconcile ctrl.Result, err error) {
+func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	r.Log.Info("Start reconciling IBU", "name", req.NamespacedName)
-	defer func() {
-		if nextReconcile.RequeueAfter > 0 {
-			r.Log.Info("Finish reconciling IBU", "name", req.NamespacedName, "requeueAfter", nextReconcile.RequeueAfter.Seconds())
-		} else {
-			r.Log.Info("Finish reconciling IBU", "name", req.NamespacedName, "requeueRightAway", nextReconcile.Requeue)
-		}
-	}()
 
-	nextReconcile = doNotRequeue()
+	// default to not requeue
+	nextReconcile := doNotRequeue()
 
 	if req.Name != utils.IBUName {
-		return
+		msg := fmt.Errorf("invalid IBU name, it must be named '%s'", utils.IBUName)
+		return doNotRequeue(), msg
 	}
 
 	ibu := &lcav1alpha1.ImageBasedUpgrade{}
-	err = r.Get(ctx, req.NamespacedName, ibu)
+	err := r.Get(ctx, req.NamespacedName, ibu)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			err = nil
-			return
+		if k8serrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			return doNotRequeue(), nil
 		}
 		r.Log.Error(err, "Failed to get ImageBasedUpgrade")
-		return
+		return requeueWithError(err)
 	}
 
 	r.Log.Info("Loaded IBU", "name", req.NamespacedName, "version", ibu.GetResourceVersion(), "desired stage", ibu.Spec.Stage)
 
+	// main
 	currentInProgressStage := utils.GetCurrentInProgressStage(ibu)
+
 	if currentInProgressStage != "" {
 		nextReconcile, err = r.handleStage(ctx, ibu, currentInProgressStage)
 		if err != nil {
-			return
+			return requeueWithError(err)
```

Comment:
This is just to make it more descriptive? Personally I prefer more concise code with named return vars.

Reply:
Yeah agreed named func parameter with no return values are cleaner. But it's only really good for small functions and large functions like our reconcile it was getting pretty tricky to follow without being explicit. 

Let me know if you like me revert it no worries!

Reply:
I see that you also removed the defer block. That was the main reason why I did that way for TALM. There were too many returns with all sorts of requeueAfter values and I didn't want to duplicate the logging lines so many times. This one is not too bad yet. Hope it stays that way :)

Reply:
Reverted!